This ghetto patch attempts to fix shared folder automounting for guests running
Linux 4.16, and also suggests an alternative to mount.vboxsf.

--- a/src/VBox/Additions/common/VBoxService/VBoxServiceAutoMount.cpp
+++ b/src/VBox/Additions/common/VBoxService/VBoxServiceAutoMount.cpp
@@ -452,6 +452,13 @@ static int vbsvcAutoMountSharedFolderOld
                       "vboxsf",
                       fFlags,
                       &mntinf);
+        if (r == -1 && errno == EINVAL)
+        {
+            /* Mainline vboxsf accepts regular mount opts. */
+            char mount_opts[1024];
+            snprintf(mount_opts, 1024, "gid=%d,dmode=0770,fmode=0770", mntinf.gid);
+            r = mount(pszShareName, pszMountPoint, "vboxsf", fFlags, mount_opts);
+        }
         if (r == 0)
         {
             VGSvcVerbose(0, "vbsvcAutoMountWorker: Shared folder '%s' was mounted to '%s'\n", pszShareName, pszMountPoint);
